backend/typescript/services/implementations/userService.ts

```diff
@@ -234,26 +241,28 @@ class UserService implements IUserService {
       }
 
       if (!oldUser) {
+        // check after re-assigning oldUser
         throw new Error(`userId ${userId} not found.`);
       }
 
       if (user.email) {
         try {
-          updatedFirebaseUser = await firebaseAdmin
+          await firebaseAdmin
             .auth()
             .updateUser(oldUser.authId, { email: user.email });
         } catch (error) {
           // rollback MongoDB user updates
           try {
-            if (user.role === "CampLeader") {
+            if (user.role === "CampLeader" || oldUser.role === "CampLeader") {
```

Comment:
just a note: when we update this in the future to change the discriminator key on role update, if `user.role && user.role !== oldUser.role`, we'll have to change the discriminator key back to the old one as well

Reply:
true...would this fall under the other ticket for adding/removing that discriminator key stuff when changing roles?

Reply:
yesss